### DISCUSSION TITLE
Multiple locations for a person

### DISCUSSION THREAD

**Discussion by oyvindsolstad** [javalent/obsidian-leaflet#100] [General]
I'm using Obsidian to get a better understanding of my **ancestors**.

So i've made a note for each of them (work in progress), and for each person I have added a "location". I had to have a standard so I picked "place of birth" - to have things consistent. 

BUT - a person typically moves about. And in some centuries people moved a lot. And far. There are just as many Americans who are descendants from Norwegians, as there are Norwegians (about five million for both). 

So it would be interesting to have a map on EACH person, showing important places in their lives. And also collect all these places in one huge map, showing all my ancestors with all their life events.

In Font Awesome there are lovely icons suitable for this:
• baby (kids)
• heart (fall in love)
• suitcase (when someone relocates)
• ring (marry)
• death (die)
• person (born)

etc. 

And I could use colors too, to make the map more readable (ie. red heart, black cross …)

So I would like to discuss how to do this best, without having the need to create separate notes for each event? 

For some of the places I've added a note for locations. So making a note for each locations _could_ work for a single person. But when I'm going to show all locations for all ancestors on a map, this method won't work, as map then would display the places, and not the persons (which SOMETIMES is better, when lots of events happen on the same place). 

And thoughts, comments, suggestions?

**Comment by valentine195** [comment]
Interesting concept! Should be easy to implement. 

I think the best way to do this is similar to the `mapoverlay` YAML tag - maybe `mapmarkers`, since `mapmarker` is taken?

**Reply by valentine195** [reply]
This also makes me think that we could give markers descriptions that could display in the tooltip, with the link (note or external) in parentheses. 

So for instance, you could have a person note (Person.md) with this frontmatter:

```
---
location: [50,50]
mapmarker: person
mapmarkers:
  - birth, [25, 25], Born
  - death, [51, 51], Died
---
```

And it would create 3 markers, and if you hovered over born it would show:

Born (Person) in the tooltip. 

If you have note preview on and the marker linked to a note, this tooltip description wouldn't work.

**Reply by oyvindsolstad** [reply]
That looks like a wonderful way to do it! And so easy to edit.

**Reply by oyvindsolstad** [reply]
This would be especially useful if the mapmarkers are visible on maps using tags and linksTo and linksFrom

Example: 

I make a note called "my ancestors" with a map that shows all persons linked from this. The map would then show all locations/markers from all persons linked to - both "location" and the other markers.

**Reply by valentine195** [reply]
Yep, the linkTo, linksFrom, markerFile, markerFolder, and markerTags parameters just tell the plugin _where_ to look for notes - it then uses the frontmatter of those notes to decide what to make.

**Reply by valentine195** [reply]
I added this exact syntax.

**Reply by oyvindsolstad** [reply]
Yey! I'll check it out!

**Comment by Moonbase59** [comment]
And I wondered if it was a typo when seeing this in the code, haha!
Real interesting use case, @oyvindsolstad! :+1:

**Reply by oyvindsolstad** [reply]
@Moonbase59 Thanks! I'm guessing it could be used for journeys too, by making the stops on an expedition or similar being mapmarkers - instead of having to make notes for each. Fiddling around to see how this works.
